src/Microsoft.SqlTools.Hosting/Hosting/Protocol/MessageDispatcher.cs

```diff
@@ -126,8 +126,14 @@ public void Stop()
                     TParams typedParams = default(TParams);
                     if (requestMessage.Contents != null)
                     {
-                        // TODO: Catch parse errors!
-                        typedParams = requestMessage.Contents.ToObject<TParams>();
+                        try
```

Comment:
previously the STS will crash if the deserialization of request parameter fails, I am catching the error here to stop it from crashing and also send the error back to ADS.